lib/internal/buffer.js

```diff
@@ -1053,6 +1053,14 @@ function markAsUntransferable(obj) {
   obj[untransferable_object_private_symbol] = true;
 }
 
+// This simply checks if the object is marked as untransferable and doesn't
+// check the ability to be transferred.
+function isMarkedAsUntransferable(obj) {
+  if ((typeof obj !== 'object' && typeof obj !== 'function') || obj === null)
```

Comment:
Nit: this can probably just be `obj?.[untransferable_object_private_symbol] === true`?

Reply:
The semantics are not exactly the same. Consider the following script:

```js
const { markAsUntransferable, isMarkedAsUntransferable } = require('worker_threads');

markAsUntransferable(Boolean.prototype);
isMarkedAsUntransferable(false); // return true if their prototype is accessed.
```

Primitive values are automatically wrapped in this case and their prototypes are checked.

However, we should strictly align the behavior between `isMarkedAsUntransferable` and the native `v8::Object::HasPrivate` -- checks the existence of the own property only. I've added a test to verify this behavior.

Thanks for the suggestion!